quattor/types/metadata.pan

```diff
@@ -4,6 +4,10 @@ type structure_branch = {
     "author" ? string
     "name" : string
     "type" : string
+    @{a unique id, e.g. hash of last git commit}
+    "id" ? string
```

Comment:
I guess the problem is that this not the id of the branch, as suggested by the namespace. While I see your use case, I am not clear how someone would actually use this: if the commit id is set inserted by the compile then there needs to be something that links that action to this field. 

Having said that, if you are doing some git hook magic to insert this without using a compiler feature, then given this is within the branch subtree, and branches are an artefact of version control, commit-id would probably be a reasonable name.